azure_arc_k8s_jumpstart/aksengine/arc_k8s_aksengine.azcli

```diff
@@ -0,0 +1,432 @@
+    --api-model $aksengine_cluster_file \
+    --client-id $sp_app_id \
+    --client-secret $sp_app_secret \
+    --set servicePrincipalProfile.clientId=$sp_app_id \
+    --set servicePrincipalProfile.secret="$sp_app_secret" \
+    --output-directory $output_dir
+
+# There are different ways to access the cluster
+# Exporting the KUBECONFIG variable is required by the command "az k8s-configuration create"
+export KUBECONFIG="$output_dir/kubeconfig/kubeconfig.$location.json" 
+kubectl get node
+# The alias is a little dirty trick for lazy persons like me
+# alias ke="kubectl --kubeconfig ./_output/$domain/kubeconfig/kubeconfig.$location.json" 
+
+
+############################
+# Enabling cluster for Arc #
+############################
+
+# Docs: https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster
+
+# Az CLI extension connectedk8s
+extension_name=connectedk8s
+extension_version=$(az extension show -n $extension_name --query version -o tsv 2>/dev/null)
+if [[ -z "$extension_version" ]]
+then
+    echo "Azure CLI extension $extension_name not found, installing now..."
+    az extension add -n $extension_name
+else
+    echo "Azure CLI extension $extension_name found with version $extension_version, trying to upgrade..."
+    az extension update -n $extension_name
+fi
+extension_version=$(az extension show -n $extension_name --query version -o tsv 2>/dev/null)
+echo "Azure CLI extension $extension_name installed with version $extension_version"
+
+# Registering providers
+for provider in "Kubernetes" "KubernetesConfiguration" "ExtendedLocation"
+do
+    registration_state=$(az provider show -n "Microsoft.${provider}" --query registrationState -o tsv)
+    if [[ "$registration_state" == "Registered" ]]
+    then
+        echo "Resource Provider Microsoft.${provider} is successfully registered with status ${registration_state}"
+    else
+        echo "It seems that provider Microsoft.${provider} is not registered, registering now..."
+        az provider register --namespace "Microsoft.${provider}"
+        wait_time=30
+        registration_state=$(az provider show -n "Microsoft.${provider}" --query registrationState -o tsv)
+        while [[ "$registration_state" != "Registered" ]]
+        do
+            echo "Registration state for RP Microsoft.${provider} is still $registration_state..."
+            sleep $wait_time
+            registration_state=$(az provider show -n "Microsoft.${provider}" --query registrationState -o tsv)
+        done
+        echo "Registration state for RP Microsoft.${provider} is ${registration_state}"
+    fi
+done
+echo "All resource providers successfully registered"
+
+# Create the ARC resource
+az group create -n $arc_rg -l $location
+az connectedk8s connect --name $arc_name -g $arc_rg
+
+# Diagnostics
+az connectedk8s list -g $arc_rg -o table
+kubectl -n azure-arc get deployments,pods
+
+####################################
+# Gitops in an arc-enabled cluster #
```

Comment:
The scenarios are divided in 2 types: 
- Bootstrapping
- Day 2 Unified operations 

Since this is for a new type of cluster "aks-engine" the script should finish in bootstrapping to Azure Arc. If the user wants GitOps, ASC, Monitoring there are specific scenarios in the Jumpstart so we can make the automations more modular.

Reply:
Done!